src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/stønadsvilkår/dto/json/OppdaterVilkårsvurderingJson.kt

```diff
@@ -0,0 +1,56 @@
+package no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.dto.json
+
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.brukerfeilHvis
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Delvilkår
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Vurdering
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.dto.DelvilkårDto
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.dto.tilDto
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.hovedreglerPassBarn
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.regler.RegelId
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.regler.SvarId
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.vilkårsreglerPassBarn
+import java.util.UUID
+
+data class OppdaterVilkårsvurderingJson(
+    val id: UUID,
+    val behandlingId: UUID,
+    val vurdering: List<VurderingJson>,
+)
+
+data class VurderingJson(
+    val regel: RegelId,
+    val svar: SvarId?,
+    val begrunnelse: String?,
+)
+
+fun VurderingJson.tilDomene(): Vurdering = Vurdering(
+    regelId = this.regel,
+    svar = this.svar,
+    begrunnelse = this.begrunnelse,
+)
+
+fun List<VurderingJson>.tilDelvilkårDtoer(): List<DelvilkårDto> {
+    val resultat = mutableListOf<DelvilkårDto>()
+
+    val hovedregler = hovedreglerPassBarn()
+    val stønadsregler = vilkårsreglerPassBarn()
+
+    for (regelId in hovedregler) {
+        val hovedregelsvar = this.find { it.regel == regelId }!!.tilDomene()
+
+        val svar = hovedregelsvar.svar
+        brukerfeilHvis(svar == null) { "Alle hovedregler må ha et svar." }
+
+        val oppfølgingsregel = stønadsregler[regelId]?.svarMapping?.get(svar)?.regelId
+
+        if (oppfølgingsregel == RegelId.SLUTT_NODE) {
+            resultat.add(Delvilkår(vurderinger = listOf(hovedregelsvar)).tilDto())
+        } else {
+            val oppfølgingssvar = this.find { it.regel == oppfølgingsregel }!!.tilDomene()
+
+            resultat.add(Delvilkår(vurderinger = listOf(hovedregelsvar, oppfølgingssvar)).tilDto())
+        }
+    }
+
+    return resultat
```

Comment:
Vid feil (`!!`) så kan det være fint å gi litt context til feilet
Disse feilene skal ju egentligen aldri skje, og då tenker jeg `error` er greit å bruke
Litt usikker på om det er forventet at frontend skal sende inn uten svar? Hvis det er det man kan bruke `ApiFeil` som er det som brukes for `brukerfeilHvis`

```suggestion
    val vurderinger = this

    val hovedregler = hovedreglerPassBarn()
    val stønadsregler = vilkårsreglerPassBarn()

    return hovedregler.map { hovedregel ->
        val hovedregelsvar = vurderinger.find { it.regel == hovedregel }?.tilDomene()
            ?: error("Finner ikke vurdering for hovedregel=$hovedregel")
        val svar = hovedregelsvar.svar ?: error("$hovedregel mangler svar")

        val oppfølgingsregel = stønadsregler[hovedregel]?.svarMapping?.get(svar)?.regelId

        if (oppfølgingsregel == RegelId.SLUTT_NODE) {
            Delvilkår(vurderinger = listOf(hovedregelsvar)).tilDto()
        } else {
            val oppfølgingssvar = vurderinger.find { it.regel == oppfølgingsregel }?.tilDomene()
                ?: error("Finner ikke svar for regel=$oppfølgingsregel")

            Delvilkår(vurderinger = listOf(hovedregelsvar, oppfølgingssvar)).tilDto()
        }
    }
```